### PR TITLE
chore(instrumentation-hapi): use exported strings for attributes

### PR DESCRIPTION
**Pull request by maryliag**
## Which problem is this PR solving?

- Part Of #2025 

## Short description of the changes

On package `opentelemetry-instrumentation-hapi`:
- Update `@opentelemetry/semantic-conventions` `^1.23.0`
- Use exported strings for Semantic Attributes.

### CODE REVIEW

**Review by trentm — CHANGES_REQUESTED** [review]
Whoa, what's going on with the 19k line change to package-lock.json?

**Comment by maryliag** [comment]
I hadn't even noticed that... what... how... but... why  😅

**Comment by trentm** [comment]
Some of the changes like:

```
     "node_modules/@aws-crypto/ie11-detection": {
       "version": "2.0.2",
-      "resolved": "https://registry.npmjs.org/@aws-crypto/ie11-detection/-/ie11-detection-2.0.2.tgz",
-      "integrity": "sha512-5XDMQY98gMAf/WRTic5G++jfmS/VLM0rwpiOpaainKi4L0nqWMSB1SzsrEG5rjFZGYN6ZAefO+/Yta2dFM0kMw==",
       "dev": true,
+      "license": "Apache-2.0",
       "dependencies": {
         "tslib": "^1.11.1"
       }
     },
```

Happen sometime for reasons I don't yet understand. I am beyond annoyed that `npm` is so fluid with its package-lock included fields that you can sometimes get monster changes like this. I haven't been able to attribute this inclusion/exclusion of fields like "license", "integrity", "resolved" to a particular npm version, or whether there was a local cache, or the order of running `npm ci` or `npm install` or `npm update` or ... I don't know.


Part of the change updating 1.24.0 to 1.24.1 and 0.51.0 to 0.51.1 are handled in https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2180

My total guess is that you ran some sort of `npm update` or related in your working tree.
I think you'll need to regenerate the package-lock change.

**Comment by maryliag** [comment]
oh yes, I did run `npm install` after updating the version, my response was more that I didn't expect that amount of change from one version change. I have a feeling no one has run a install or update on this package for awhile.
Something on a new version is breaking one of the tests, so I would also need to check that

**Comment by maryliag** [comment]
I merged the main branch since the PR you mentioned was merged, and the lock file went form 19k to 17k, which is still a lot, but maybe this is just expected and we should merge it (after I figure out the test failure)

**Comment by JamieDanielson** [comment]
> I merged the main branch since the PR you mentioned was merged, and the lock file went form 19k to 17k, which is still a lot, but maybe this is just expected and we should merge it (after I figure out the test failure)

I believe the test failure is related to the package-lock change, and if that can get regenerated then the tests will pass.

Locally I pulled down main, went to the hapi directory, and ran `npm i @opentelemetry/semantic-conventions@1.22.0` and then `npm run compile`. Then from the root directory I ran `npm ci`, `npm run compile`, and `npm run test`. Aside from the mongoose test timeout errors I always get locally and the aws test error (new? unrelated and not a problem in CI), other tests passed and my package-lock change is minimal.

**Comment by trentm** [comment]
I pushed a small tweak to package-lock to get instr-hapi to use v1.24.1 in the package-lock, even though it is `^1.22.0` in the package.json. I did this by running `npm update @opentelemetry/semantic-conventions` in the instr-hapi dir.

File: plugins/node/opentelemetry-instrumentation-hapi/README.md [hunk 1]
```diff
@@ -66,6 +66,10 @@ The dev dependency of `@hapi/podium@4.1.1` is required to force the compatible t
 
 This package provides automatic tracing for hapi server routes and [request lifecycle](https://github.com/hapijs/hapi/blob/master/API.md#request-lifecycle) extensions defined either directly or via a Hapi plugin.
 
+## Semantic Conventions
+
+This package uses `@opentelemetry/semantic-conventions` version `1.22+`, which implements Semantic Convention [Version 1.7.0](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.7.0/semantic_conventions/README.md)
+
```
Comment (trentm) [inline plugins/node/opentelemetry-instrumentation-hapi/README.md:72]: I guess it would be nice to have a table of used attrs here?

Comment (maryliag) [reply plugins/node/opentelemetry-instrumentation-hapi/README.md:72]: done

**Review by trentm — APPROVED** [review]

**Pull request merged**